src/lib/messenger/scroll.ts

```diff
@@ -38,13 +38,14 @@ const domRectToRect = (rect: DOMRect) => ({
 });
 
 const sendCoordinates = (iframeId: string, domRect: DOMRect) => {
-	iframes[iframeId].respond(null, domRectToRect(domRect));
+	iframes[iframeId]?.respond(null, domRectToRect(domRect));
 };
 
-const getDimensions = (id: string): [string, DOMRect] => [
-	id,
-	iframes[id].node.getBoundingClientRect(),
-];
+const getDimensions = (id: string): [string, DOMRect] => {
+	// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion  -- Fixes a type error caused by the optional chain
+	const node = <HTMLIFrameElement>iframes[id]?.node;
```

Comment:
Just to confirm my understanding, are we assuming that `iframes[id]` is always defined here, as we're casting to tell TypeScript it'll never be undefined?

Reply:
Yes we are - I'm not thrilled about handling it this way, but allowing the function to return undefined causes more type errors downstream where getDimensions is used. Very open to nicer ideas of how we can handle it!

Reply:
No it's all good, just wanted to check!